docs/choreolib/auto-factory.md

```diff
@@ -1,109 +1,225 @@
+    scoreTraj.active().whileTrue(scoringSubsystem.getReady());
+
+    // When the trajectory is done, score
+    scoreTraj.done().onTrue(scoringSubsystem.score());
 
     return routine;
-  }
+}
+```
+
+1. You should always reset your robot's odometry to the start of the first trajectory being followed in an autonomous routine. `AutoTrajectory.resetOdometry()` will accomplish this, setting the robot's pose to the start of the trajectory.
+
+Sometimes, you may want to implement a "branching auto": an autonomous routine that changes behavior based on the state of the robot. An excellent example for the need of branching autos is the [2024 season](https://youtu.be/9keeDyFxzY4), where robots would race to the midline to grab a gamepiece. If another robot beat yours to the midline, or your robot missed a game piece, a common strategy was to go directly to the next gamepiece on the midline, instead of coming back to score.
+
+Below is an example auto routine from the 2024 season:
 
-  public void autonomousInit() {
-    autoChooser.getSelectedAutoRoutine().schedule();
-  }
+```java
+public AutoRoutine branching2024Auto() {
+    AutoRoutine routine = autoFactory.newRoutine("branching2024Auto");
+
+    // This routine uses segments between pre-defined handoff points.
+    // Expand the tooltip for information about their naming convention -> (1)
+
+    // Load the routine's trajectories
+    AutoTrajectory startToC2 = routine.trajectory("startToC2");
+    AutoTrajectory C2toM1 = routine.trajectory("C2toM1");
+    AutoTrajectory M1toM2 = routine.trajectory("M1toM2");
+    AutoTrajectory M2toM3 = routine.trajectory("M2toM3");
+    AutoTrajectory M1toScore = routine.trajectory("M1toScore");
+    AutoTrajectory M2toScore = routine.trajectory("M2toScore");
+    AutoTrajectory M3toScore = routine.trajectory("M3toScore");
+    AutoTrajectory scoreToM2 = routine.trajectory("scoreToM2");
+    AutoTrajectory scoreToM3 = routine.trajectory("scoreToM3");
+
+    // When the routine starts, reset odometry, shoot the first gamepiece, then go to the "C2" location
+    routine.running().onTrue(
+        Commands.sequence(
+            startToC2.resetOdometry(),
+            shooterSubsystem.shoot(),
+            startToC2.cmd()
+        )
+    );
+
+    // Pick up and shoot the gamepiece at the "C2" location, then go to the "M1" location
+    startToC2.active().whileTrue(intakeSubsystem.intake());
+    startToC2.done().onTrue(shooterSubsystem.shoot().andThen(C2toM1.cmd()));
+
+    // Run the intake when we are approaching a gamepiece
+    C2toM1.active()
+        .or(scoreToM2.active())
+        .or(scoreToM3.active())
+        .or(M1toM2.active())
+        .or(M2toM3.active())
+        .whileTrue(intakeSubsystem.intake());
+
+    // If we picked up the gamepiece, go to the "Score" location
+    // If we didn't pick up the gamepiece, go to the next midline location
+    C2toM1.done().onTrue(either(M1toScore.cmd(), M1toM2.cmd(), shooterSubsystem::hasGamepiece));
```

Comment:
maybe not here but somewhere note that either requires the subsystems of both branches, if you want each branch to be completely separate you can do

```java
C2toM1.done().and(shooterSubsystem::hasGamepiece).onTrue(M1toScore.cmd());
C2toM1.done().and(shooterSubsystem::noGamepiece).onTrue(M1toM2.cmd());
```
or smthn, this isn't the best example of where it might matter but you get the idea

Reply:
I'll replace with the modified code and add a note dialog (blue box) about the requirements nuance